### ISSUE TITLE
Using the selected value

### ISSUE THREAD

**Issue by diogenesbrussels** [ConorWebb96/bb-type-ahead#2]
Hey @ConorWebb96 ,

I'm having trouble getting the value selected by the field.
Here is my configuration:
![image](https://github.com/ConorWebb96/bb-type-ahead/assets/91942877/f9ea7c4e-e9b1-4f88-afdd-ae4e2fb275a4)
![image](https://github.com/ConorWebb96/bb-type-ahead/assets/91942877/f922d35d-cca1-4ca1-8dfb-446a3b8d5c78)

In the component action, I want to change the value of another field to that of the selected element (the selected id into a number field).
I've tried the following:
```
{{ Field Value }}
{{ Field Value.value }}
{{ Field Value }}.value
return $("Field Value");
return $("Field Value").value;
return $("Field Value.value");
```

What am i missing? thank you!

**Comment by ConorWebb96** [comment]
Hey @diogenesbrussels,

Default values won't work on other fields in Budibase, they will only work if you set/have the value before page load. I'd suggest using the `on change` button and then adding a `Update field value` action within the component to then set the value of your `otherHabitant` field.

Due to your question, I noticed an issue were it will show the value in array form even though it should be string based. I'll add a fix tomorrow to so that it only displays the selected item as a string. (This will only happen when the type is set to fields, otherwise it will output the values out in the Budibase array format.)
<img width="299" alt="Screenshot 2023-07-05 at 15 20 23" src="https://github.com/ConorWebb96/bb-type-ahead/assets/126772285/7bd29560-f66a-435b-93fc-c990eba06585">

You will likely need to do something like this current inside the `Update field value` action: `{{ Field Value.[0].value }}`

Below is a demo of this working.
![demo updating field with type ahead](https://github.com/ConorWebb96/bb-type-ahead/assets/126772285/ee031bc4-2681-44bf-bc95-90efd2b03044)

I hope this makes sense and is helpful.

**Comment by diogenesbrussels** [comment]
Thank for the quick reply!

I was already using the onChange event but didn't know how to access the value.
But {{ Field Value.[0].value }} works well, I didn't think about it because I confused {{ for [{
Sometimes the obvious is hard to see. I wondered about the return of multiple choices without thinking any further :D

**Comment by ConorWebb96** [comment]
Hey @diogenesbrussels,

I've added a fix for this here: https://github.com/ConorWebb96/bb-type-ahead/pull/3 with the new version whenever you are using the string type you should now only need to use `{{ Field Value }}` instead of drilling into the array to get the value.

Sorry for the delay has been a busy few days!

**Comment by diogenesbrussels** [comment]
> Hey @diogenesbrussels,
> 
> I've added a fix for this here: #3 with the new version whenever you are using the string type you should now only need to use `{{ Field Value }}` instead of drilling into the array to get the value.
> 
> Sorry for the delay has been a busy few days!

Thank you for all your work!
